e3nn/o3/_irreps.py

```diff
@@ -353,13 +359,13 @@ def __new__(cls, irreps=None):
                 elif len(mul_ir) == 3:
                     mul, l, p = mul_ir
                     ir = Irrep(l, p)
-                    warnings.warn("prefer using [(mul, (l, p))] to distinguish multiplicity from irrep", DeprecationWarning, stacklevel=2)
+                    warnings.warn("Use the argument format [(mul, (l, p)), ...] to distinguish multiplicity from irrep", DeprecationWarning, stacklevel=2)
                 else:
                     mul = None
                     ir = None
 
-                assert isinstance(mul, int) and mul >= 0
-                assert ir is not None
+                if not (isinstance(mul, int) and mul >= 0 and ir is not None):
+                    raise ValueError(f"Unable to interpret \"{mul_ir}\" as an irrep.")
```

Comment:
Raising a `ValueError` here is probably more Pythonic than an assertion. Assertions can be disabled in "optimized" Python and aren't as informative as an exception like `ValueError`, `TypeError`, etc. I did this in a couple other places, just wanted to explain why. 👍